abstract_commands.xml

```diff
@@ -5,8 +5,26 @@
         is accessible, then \Fsize matching the register's actual size must be
         supported.
 
+        \begin{steps}{Perform the following sequence of operations:}
+        \item Halt the hart, if \Fprehalt is set.
+        \item Execute the Program Buffer, if \Fpreexec is set.
+        \item Copy data from the specified register into {\tt data}, if \Fwrite is clear.
+        \item Copy data from {\tt data} into the specified register, if \Fwrite is set.
+        \item Execute the Program Buffer, if \Fpostexec is set.
+        \item Resume the hart, if \Fpostresume is set.
+        \end{steps}
+
         <field name="0" bits="31:24" />
-        <field name="0" bits="23:22" />
+        <field name="prehalt" bits="23">
+            When 1, halt the hart before performing the rest of the command.
```

Comment:
Yes. This allows quickly accessing the hart. The use case is eg. PC sampling, or doing some kind of debugging on systems that shouldn't be halted because they have real-time requirements (but not so hard that they can't spare a hundred cycles here and there).

Quickly doing breakpoint->code->resume is an interesting idea, but I'm not sure how to implement it. Eg. what happens if all your harts hit the breakpoint at the same time? Making the DM handle them all in sequence starts to take a lot of time, and get a lot more complicated. What if the breakpoint hits again in the first hart when you're still going through the list? Can the debugger use the Program Buffer while something like this is set up? Maybe it's sufficient to handle those corner cases poorly. My inclination is to require the software to be instrumented if you want something like that. The serial ports could be used to get data out.